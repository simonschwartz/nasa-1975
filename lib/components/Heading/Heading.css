.NASA1975-heading {
  font-family: "Helvetica", "Segoe UI", Arial, sans-serif;
  margin: 0 0 1rem 0;
}

.NASA1975-heading--xl {
  font-size: 2.5rem;
  line-height: 1.25;
}

.NASA1975-heading--lg {
  font-size: 2rem;
  line-height: 1.3;
}

.NASA1975-heading--md {
  font-size: 1.75rem;
  line-height: 1.25;
}

.NASA1975-heading--sm {
  font-size: 1.25rem;
  line-height: 1.33333;
}

.NASA1975-heading--xs {
  font-size: 1rem;
  line-height: 1.2;
}

.NASA1975-heading--medium {
  font-weight: 600;
}

.NASA1975-heading--light {
  font-weight: 300;
}

@media (min-width: 768px) {
  .NASA1975-heading--xl {
    font-size: 3rem;
  }

  .NASA1975-heading--lg {
    font-size: 2.5rem;
  }

  .NASA1975-heading--md {
    font-size: 2rem;
  }

  .NASA1975-heading--sm {
    font-size: 1.5rem;
  }

  .NASA1975-heading--xs {
    font-size: 1.25rem;
  }
}
