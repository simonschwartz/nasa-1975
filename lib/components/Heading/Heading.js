import React from 'react';
import PropTypes from 'prop-types';
import { cx, css } from 'emotion';
import { fontFamily } from '../../styles';
import './Heading.css';

const headingBase = css`
  margin: 0 0 1rem 0;
  font-family: ${fontFamily}
`;

const fontSizeXL = css`
  font-size: 2.5rem;
  line-height: 1.25;

  @media (min-width: 768px) {
    font-size: 3rem;
  }
`;

const Heading = ({ size = 'lg', level = '1', weight = 'medium', children, className = '', ...attributeOptions }) => {
  const HeadingTag = `h${level}`;
  return (
    <HeadingTag
      className={cx(headingBase)}
      className={`NASA1975-heading NASA1975-heading--${size} NASA1975-heading--${weight} ${className}`}
      {...attributeOptions}
    >
      {children}
    </HeadingTag>
  );
};

Heading.propTypes = {
  /** Size of the heading */
  size: PropTypes.oneOf(['xs', 'sm', 'md', 'lg', 'xl']),
  /** HTML heading level */
  level: PropTypes.oneOf(['1', '2', '3', '4', '5']),
  /** Font weight of heading level */
  weight: PropTypes.oneOf(['medium', 'light']),
  /** Heading content */
  children: PropTypes.node.isRequired,
  /** className to add to heading */
  className: PropTypes.string
};

export default Heading;
