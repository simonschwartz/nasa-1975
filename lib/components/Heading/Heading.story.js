import React from "react";
import { text, select } from "@storybook/addon-knobs";
import { withInfo } from "@storybook/addon-info";
import Heading from "./";

const HeadingStory = withInfo({
  text:
    "Headings are set in Helvetica Medium. In certain situations Helvetica Bold may be an appropriate alternative."
})(() => {
  // Setup knob for setting heading size
  const labelSize = "size";
  const optionsSize = {
    xl: "xl",
    lg: "lg",
    md: "md",
    sm: "sm",
    xs: "xs"
  };
  const defaultValueSize = "md";
  const sizeValue = select(labelSize, optionsSize, defaultValueSize);

  // Setup knob for setting heading level
  const labelLevel = "level";
  const optionsLevel = {
    1: "1",
    2: "2",
    3: "3",
    4: "4",
    5: "5"
  };
  const defaultValueLevel = "1";
  const levelValue = select(labelLevel, optionsLevel, defaultValueLevel);

  // Setup knob for setting heading level
  const labelWeight = "weight";
  const optionsWeight = {
    light: "light",
    medium: "medium",
    bold: "bold"
  };
  const defaultValueWeight = "bold";
  const weightValue = select(labelWeight, optionsWeight, defaultValueWeight);

  // Setup knob for setting header text
  const labelText = "text";
  const defaultValueText = "Heading value";
  const textValue = text(labelText, defaultValueText);

  return (
    <Heading size={sizeValue} level={levelValue} weight={weightValue}>
      {textValue}
    </Heading>
  );
});

export default HeadingStory;
