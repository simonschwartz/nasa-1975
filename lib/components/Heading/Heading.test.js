import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import Heading from "./";

it("renders correct markup with no props", () => {
  const tree = renderer.create(<Heading>Test heading text</Heading>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correct markup with size prop", () => {
  const tree = renderer
    .create(<Heading size="md">Test heading text</Heading>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correct markup with level prop", () => {
  const tree = renderer
    .create(<Heading level="3">Test heading text</Heading>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correct markup with weight prop", () => {
  const tree = renderer
    .create(<Heading weight="light">Test heading text</Heading>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correct markup with multiple props", () => {
  const tree = renderer
    .create(
      <Heading size="xs" level="4" weight="light">
        Test heading text
      </Heading>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Heading>Test heading text</Heading>, div);
});
