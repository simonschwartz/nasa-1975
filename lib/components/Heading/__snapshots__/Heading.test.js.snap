// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correct markup with level prop 1`] = `
<h3
  className="NASA1975-heading NASA1975-heading--lg NASA1975-heading--medium "
>
  Test heading text
</h3>
`;

exports[`renders correct markup with multiple props 1`] = `
<h4
  className="NASA1975-heading NASA1975-heading--xs NASA1975-heading--light "
>
  Test heading text
</h4>
`;

exports[`renders correct markup with no props 1`] = `
<h1
  className="NASA1975-heading NASA1975-heading--lg NASA1975-heading--medium "
>
  Test heading text
</h1>
`;

exports[`renders correct markup with size prop 1`] = `
<h1
  className="NASA1975-heading NASA1975-heading--md NASA1975-heading--medium "
>
  Test heading text
</h1>
`;

exports[`renders correct markup with weight prop 1`] = `
<h1
  className="NASA1975-heading NASA1975-heading--lg NASA1975-heading--light "
>
  Test heading text
</h1>
`;
